python/pyspark/sql/tests/streaming/test_streaming_foreach_family.py

```diff
@@ -0,0 +1,369 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import tempfile
+import time
+
+from pyspark.testing.sqlutils import ReusedSQLTestCase
+
+
+class StreamingTestsForeachFamilyMixin:
+    class ForeachWriterTester:
```

Comment:
When I tried these tests, I got a serialization error after moving this to 'Mixin'. We could handle it when run into it with connect tests.

Reply:
I get all tests passed by running ` python/run-tests --testnames pyspark.sql.tests.streaming.test_streaming_foreach_family`. I renamed the mixin part though in the new commit

Reply:
Can we move `foreach` back to its original place. Foreach() and foreachbath() are very different. The later takes a dataframe, the former is more like a UDF.

Reply:
Please add foreach() api to one of the jiras or a new jira.